### PR TITLE
Rename class: ApiKeyUserData->UserWithRoleRecord

### PR DESCRIPTION
**Pull request by DanVanAtta**
## Functional Changes

[] New map or map update
[] New Feature
[] Feature update or enhancement
[] Feature Removal
[x] Code Cleanup or refactor
[] Configuration Change
[] Bug fix:  
[] Other:   

## Testing

[x] No manual testing done
[] Manually tested

### CODE REVIEW

File: http-server/src/main/java/org/triplea/server/lobby/chat/ChatParticipantAdapter.java [hunk 1]
```diff
@@ -3,12 +3,12 @@
...
 import org.triplea.http.client.lobby.chat.ChatParticipant;
-import org.triplea.lobby.server.db.data.ApiKeyUserData;
 import org.triplea.lobby.server.db.data.UserRole;
+import org.triplea.lobby.server.db.data.UserWithRoleRecord;
 
-public class ChatParticipantAdapter implements Function<ApiKeyUserData, ChatParticipant> {
+public class ChatParticipantAdapter implements Function<UserWithRoleRecord, ChatParticipant> {
   @Override
-  public ChatParticipant apply(final ApiKeyUserData apiKeyUserData) {
+  public ChatParticipant apply(final UserWithRoleRecord apiKeyUserData) {
```
Comment (RoiEXLab) [inline http-server/src/main/java/org/triplea/server/lobby/chat/ChatParticipantAdapter.java]: The parameter could be renamed as well

File: http-server/src/test/java/org/triplea/server/access/ApiKeyAuthenticatorTest.java [hunk 2]
```diff
@@ -13,22 +13,26 @@
...
 
-  private static final ApiKeyUserData HOST_DATA =
-      ApiKeyUserData.builder().role(UserRole.HOST).build();
+  private static final UserWithRoleRecord HOST_DATA =
+      UserWithRoleRecord.builder().role(UserRole.HOST).build();
 
-  private static final ApiKeyUserData ANONYMOUS_USER_DATA =
-      ApiKeyUserData.builder().username("anonymous-user-name").role(UserRole.ANONYMOUS).build();
+  private static final UserWithRoleRecord ANONYMOUS_USER_DATA =
+      UserWithRoleRecord.builder().username("anonymous-user-name").role(UserRole.ANONYMOUS).build();
```
Comment (RoiEXLab) [inline http-server/src/test/java/org/triplea/server/access/ApiKeyAuthenticatorTest.java]: I mean it's still data, but I'd rename the `_DATA` suffix to `_RECORD` for those 3 variables

File: lobby-db-dao/src/main/java/org/triplea/lobby/server/db/data/UserWithRoleRecord.java [hunk 3]
```diff
@@ -28,10 +28,10 @@
   private String role;
 
   /** Returns a JDBI row mapper used to convert a ResultSet into an instance of this bean object. */
-  public static RowMapper<ApiKeyUserData> buildResultMapper() {
+  public static RowMapper<UserWithRoleRecord> buildResultMapper() {
     return (rs, ctx) -> {
-      final ApiKeyUserData userData =
-          ApiKeyUserData.builder()
+      final UserWithRoleRecord userData =
```
Comment (RoiEXLab) [inline lobby-db-dao/src/main/java/org/triplea/lobby/server/db/data/UserWithRoleRecord.java]:
```suggestion
      final UserWithRoleRecord userRecord =
```

File: integration-testing/src/test/java/org/triplea/lobby/server/db/dao/api/key/LobbyApiKeyDaoTest.java [hunk 4]
```diff
@@ -10,24 +10,24 @@
...
           .role(UserRole.MODERATOR)
           .build();
-  private static final ApiKeyUserData EXPECTED_ANONYMOUS_DATA =
-      ApiKeyUserData.builder().username("some-other-name").role(UserRole.ANONYMOUS).build();
-  private static final ApiKeyUserData EXPECTED_HOST_DATA =
-      ApiKeyUserData.builder().role(UserRole.HOST).build();
+  private static final UserWithRoleRecord EXPECTED_ANONYMOUS_DATA =
+      UserWithRoleRecord.builder().username("some-other-name").role(UserRole.ANONYMOUS).build();
+  private static final UserWithRoleRecord EXPECTED_HOST_DATA =
+      UserWithRoleRecord.builder().role(UserRole.HOST).build();
```
Comment (RoiEXLab) [inline integration-testing/src/test/java/org/triplea/lobby/server/db/dao/api/key/LobbyApiKeyDaoTest.java]: Same here with the suffix

File: http-server/src/test/java/org/triplea/server/lobby/chat/MessagingServiceTest.java [hunk 5]
```diff
@@ -46,8 +46,11 @@
 
   private static final ChatParticipantAdapter chatParticipantAdapter = new ChatParticipantAdapter();
 
-  private static final ApiKeyUserData API_KEY_USER_DATA =
-      ApiKeyUserData.builder().role(UserRole.MODERATOR).username("player-name-moderator").build();
+  private static final UserWithRoleRecord API_KEY_USER_DATA =
+      UserWithRoleRecord.builder()
+          .role(UserRole.MODERATOR)
+          .username("player-name-moderator")
+          .build();
```
Comment (RoiEXLab) [inline http-server/src/test/java/org/triplea/server/lobby/chat/MessagingServiceTest.java]: And here as well

File: http-server/src/test/java/org/triplea/server/access/ApiKeyAuthenticatorTest.java (hunk 2 above)
Comment (DanVanAtta) [reply http-server/src/test/java/org/triplea/server/access/ApiKeyAuthenticatorTest.java]: Might be better to avoid any suffix

File: http-server/src/test/java/org/triplea/server/lobby/chat/MessagingServiceTest.java (hunk 5 above)
Comment (DanVanAtta) [reply http-server/src/test/java/org/triplea/server/lobby/chat/MessagingServiceTest.java]: Calling this `MODERATOR_DATA`, `MODERATOR_RECORD` is misleading, and probably also not the best to just name this data after its type but instead to match what it represents.

File: integration-testing/src/test/java/org/triplea/lobby/server/db/dao/api/key/LobbyApiKeyDaoTest.java (hunk 4 above)
Comment (DanVanAtta) [reply integration-testing/src/test/java/org/triplea/lobby/server/db/dao/api/key/LobbyApiKeyDaoTest.java]: I'm hesitant. `EXPECTED_ANONYMOUS_RECORD` could be thought of as a record in the lobby_user table, that the data represents an `ANONYMOUS` user rather than a `UserWithRole`.  `EXPECTED_ANONYMOUS_USER_WITH_ROLE_RECORD` is too wordy (just a poor name there where we should be able to name this after what it represents and avoid baking in the type information).

**Comment by DanVanAtta** [comment]
Took a bit to get the build to pass, added a TODO to come back to that.
